### ISSUE TITLE
Assert failure "Looks like a random breakpoint/trap that was not prepared by the EE debugger"

### ISSUE THREAD

**Issue by saper** [dotnet/coreclr#1113]
When debugging https://github.com/dotnet/coreclr/issues/1102 on FreeBSD I have managed to trigger 

````
Assert failure (unable to format)
/home/saper/git/coreclr/src/vm/stackwalk.cpp
!"Thread::VirtualUnwindToFirstManagedCallFrame: PAL_VirtualUnwind failed"
**** MessageBox invoked, title 'Assert failure (unable to format)' ****
  !"Thread::VirtualUnwindToFirstManagedCallFrame: PAL_VirtualUnwind failed"
````

(this can be reproduced by applying https://gist.github.com/f58345e94f949d69090d and trying to run a non-CLR executable via `./corerun <name>`)

After assertion I get the **endless stream** of

````
********

Assert failure(PID 17724 [0x0000453c], Thread: 29385728 [0x1c06400]): !"Looks like a random breakpoint/trap that was not prepared by the EE debugger"
    File: /home/saper/git/coreclr/src/vm/exceptionhandling.cpp, Line: 5066 Image:
/home/saper/sw/dotnet/roslyn/corerun
**** MessageBox invoked, title 'corerun - Assert Failure (PID 17724, Thread 29385728/1c06400)        ' ****
  !"Looks like a random breakpoint/trap that was not prepared by the EE debugger"

/home/saper/git/coreclr/src/vm/exceptionhandling.cpp, Line: 5066

Abort - Kill program
Retry - Debug
Ignore - Keep running
````

This should not be the case, the program should (ideally) exit.

**Comment by jasonwilliams200OK** [comment]
Since #1136 is merged, can you confirm if this is still happening with latest master?

**Comment by kangaroo** [comment]
This shouldn't be happening since it required an incorrect patch to trigger this.

**Comment by tzwlai** [comment]
Closing this issue since it looks like the problem has been addressed by #1136 and there's not been any activity in a while.  Please feel free to reopen if anyone feels differently.
